File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left = 0;
        int right = 0;
        int max_len = 0;
        unordered_map<int,int> map;

        while(s[right]){
            map[s[right]]++;
            while(map[s[right]] > 1){
                map[s[left++]]--;
            }
            max_len = max(max_len,right - left + 1);
            right++;
        }
        return max_len;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        biggest_pile = max(piles)
        curr_pace = 0
        left = 1
        right = biggest_pile

        while left <= right:
            mid = int((left+right)/2)
            if self.finish_piles(piles,mid) <= h:
                print(mid)
                curr_pace = mid
                right = mid - 1
            else:
                left = mid + 1
        
        if self.finish_piles(piles,curr_pace) <= h:
            return curr_pace
        return -1

    def finish_piles(self,piles,pace):
        hours = 0
        for pile in piles:
            if pace:
                hours += math.ceil(pile/pace)
        return hours

            


        
        